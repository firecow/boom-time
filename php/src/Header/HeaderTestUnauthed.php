<?
declare(strict_types=1);

namespace App\Header;

use App\Exceptions\DAOException;
use App\Exceptions\FileErrorException;
use App\Exceptions\FileNotFoundException;
use App\Mocks\ContextMock;
use App\Mocks\SuperGlobalsMock;
use App\Testing\RegressionTest;

class HeaderTestUnauthed extends RegressionTest
{

    /**
     * @throws FileErrorException
     * @throws FileNotFoundException
     * @throws DAOException
     */
    public function doTest(): string
    {
        $superGlobals = new SuperGlobalsMock();
        $ctx = new ContextMock($superGlobals);
        return Header::getHeaderHTML($ctx);
    }
}
