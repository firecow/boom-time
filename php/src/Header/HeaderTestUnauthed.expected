<button onclick="return simplePHP.pushPageToHistory('/games/')">
    <i class="fas fa-gamepad"></i>
    <c>Games</c>
</button>
<button onclick="return simplePHP.pushPageToHistory('/error/')">
    <i class="fas fa-exclamation-circle"></i>
    <c>Error</c>
</button>
<button onclick="return simplePHP.openModal('login')">
    <i class="fas fa-sign-in-alt"></i>
    <c>Login</c>
</button>
<button onclick="return simplePHP.openModal('signup')">
    <i class="fas fa-user-plus"></i>
    <c>Signup</c>
</button>